app/containers/Avatar.js

```diff
@@ -28,7 +29,7 @@ const Avatar = React.memo(({
 	const uriSize = size === 100 ? 100 : 50;
 
 	let avatarAuthURLFragment = '';
-	if (userId && token) {
+	if (userId && token && !Accounts_AvatarBlockUnauthenticatedAccess) {
```

Comment:
The right behavior is exactly the opposite, when `Accounts_AvatarBlockUnauthenticatedAccess` is `true` we need to add the `avatarAuthURLFragment`, if it's `false` we don't need to add this.